src/storage/leveldb/nanocontract_index.ts

```diff
@@ -0,0 +1,118 @@
+import path from 'path';
+import { Level } from 'level';
+import { AbstractSublevel } from 'abstract-level';
+import { IKVNanoContractIndex, INcData } from "src/types";
+import { errorCodeOrNull, KEY_NOT_FOUND_CODE } from './errors';
+
+export const REGISTERED_PREFIX = 'registered';
+
+export default class LevelNanoContractIndex implements IKVNanoContractIndex {
+  dbpath: string;
+  /**
+   * Registered Nano Contract database
+   * Key: nckey
+   * Value: INcData (json encoded)
+   */
+  registeredDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, INcData>;
+  indexVersion: string = '0.0.1';
+
+  constructor(dbpath: string) {
+    this.dbpath = path.join(dbpath, 'nanocontract');
+    const db = new Level(this.dbpath);
+    this.registeredDB = db.sublevel<string, INcData>(REGISTERED_PREFIX, { valueEncoding: 'json' });
+  }
+
+  /**
+   * Close the database and the sublevel children.
+   * @returns {Promise<void>}
+   */
+  async close(): Promise<void> {
+    await this.registeredDB.db.close();
+  }
+
+  /**
+   * Check that the index version matches the expected version.
+   * @returns {Promise<void>}
+   */
+  async checkVersion(): Promise<void> {
```

Comment:
This method is exactly the same as in the Token index class. Does it make sense to create an util method, or an abstract class that they extend?

Reply:
It seems to be equal in all leveldb implementations.

Reply:
Refactored here https://github.com/HathorNetwork/hathor-wallet-lib/pull/614/commits/32a8c3fc15efd3e2886aad073f1a3d0f3f6da5fa

Reply:
Much better, thanks